crates/db_schema/src/impls/person.rs

```diff
@@ -121,6 +121,16 @@ impl Person {
       .load::<CommunityId>(conn)
       .await
   }
+
+  pub async fn is_username_taken(pool: &mut DbPool<'_>, username: &str) -> Result<bool, Error> {
+    use diesel::dsl::{exists, select};
+    let conn = &mut get_conn(pool).await?;
+    select(exists(
+      person::table.filter(lower(person::name).eq(username.to_lowercase())),
+    ))
+    .get_result(conn)
+    .await
+  }
```

Comment:
this function seems pretty suspicious to me, for one because it does not check whether the exists is local (so it also checks remote instance usernames?) and secondly, because if this is a requirement then it should probably be enforced using a DB constraint instead of a check, because this kind of check is always prone to a race condition if called twice from different sources simultaneously.

Also, why is it only called for oauth registration and not normal registration? What's the reason it's needed there but not the other?

Reply:
If you add this then be consistent and call it for normal registration too.